### ISSUE TITLE
Crash on startup on OnePlus 9

### ISSUE THREAD

**Issue by olokobayusuf** [natmlx/NatDevice#8]
Apparently doesn't affect other devices. Reported by [IgorBoyko](https://forum.unity.com/members/igorboyko.5931699/).
![image](https://user-images.githubusercontent.com/17663473/193079925-c32a5f9a-25d2-4dd9-a5cc-d4aec4ca36fb.png)

**Comment by olokobayusuf** [comment]
Logs:
```
022-09-29 18:21:49.393 30923-31581/com.redacted.redacted I/BpBinder: onLastStrongRef automatically unlinking death recipients: <uncached descriptor>
2022-09-29 18:21:49.393 30923-31581/com.redacted.redacted I/BpBinder: onLastStrongRef automatically unlinking death recipients: <uncached descriptor>
2022-09-29 18:21:49.410 30923-31581/com.redacted.redacted E/AudioRecord: createRecord_l(6356): AudioFlinger could not create record track, status: -1
2022-09-29 18:21:49.410 30923-31581/com.redacted.redacted W/AudioRecord: restoreRecord_l(6356): failed status -1, retries 2
2022-09-29 18:21:49.910 30923-31581/com.redacted.redacted I/BpBinder: onLastStrongRef automatically unlinking death recipients: <uncached descriptor>
2022-09-29 18:21:49.926 30923-31581/com.redacted.redacted E/AudioRecord: createRecord_l(6356): AudioFlinger could not create record track, status: -1
2022-09-29 18:21:49.926 30923-31581/com.redacted.redacted W/AudioRecord: restoreRecord_l(6356): failed status -1, retries 1
2022-09-29 18:21:49.926 30923-31581/com.redacted.redacted E/AudioRecord-JNI: Error -1 during AudioRecord native read
2022-09-29 18:21:49.926 30923-31581/com.redacted.redacted W/libc: malloc(18446744073709551615) failed: returning null pointer
2022-09-29 18:21:49.928 30923-31581/com.redacted.redacted E/libc++abi: terminating with uncaught exception of type std::bad_alloc: std::bad_alloc
2022-09-29 18:21:50.108 30923-31581/com.redacted.redacted E/CRASH: *** *** *** *** *** *** *** *** *** *** *** *** *** *** *** ***
2022-09-29 18:21:50.108 30923-31581/com.redacted.redacted E/CRASH: Version '2021.3.5f1 (40eb3a945986)', Build type 'Development', Scripting Backend 'il2cpp', CPU 'arm64-v8a'
2022-09-29 18:21:50.108 30923-31581/com.redacted.redacted E/CRASH: Build fingerprint: 'OnePlus/OnePlus9/OnePlus9:12/RKQ1.211119.001/R.202208121853:user/release-keys'
2022-09-29 18:21:50.108 30923-31581/com.redacted.redacted E/CRASH: Revision: '0'
2022-09-29 18:21:50.108 30923-31581/com.redacted.redacted E/CRASH: ABI: 'arm64'
2022-09-29 18:21:50.108 30923-31581/com.redacted.redacted E/CRASH: Timestamp: 2022-09-29 18:21:50+0300
2022-09-29 18:21:50.108 30923-31581/com.redacted.redacted E/CRASH: pid: 30923, tid: 31581, name: Thread-24  >>> com.redacted.redacted <<<
2022-09-29 18:21:50.108 30923-31581/com.redacted.redacted E/CRASH: uid: 10399
2022-09-29 18:21:50.108 30923-31581/com.redacted.redacted E/CRASH: signal 6 (SIGABRT), code -1 (SI_QUEUE), fault addr --------
2022-09-29 18:21:50.108 30923-31581/com.redacted.redacted E/CRASH:     x0  0000000000000000  x1  0000000000007b5d  x2  0000000000000006  x3  0000007554be1470
2022-09-29 18:21:50.108 30923-31581/com.redacted.redacted E/CRASH:     x4  736f646277641f73  x5  736f646277641f73  x6  736f646277641f73  x7  7f7f7f7f7f7f7f7f
2022-09-29 18:21:50.108 30923-31581/com.redacted.redacted E/CRASH:     x8  00000000000000f0  x9  b9efe8a004ca2f92  x10 0000000000000000  x11 ffffff80fffffbdf
2022-09-29 18:21:50.108 30923-31581/com.redacted.redacted E/CRASH:     x12 0000000000000001  x13 00003215055312db  x14 00373df0a02a9fa1  x15 0000000034155555
2022-09-29 18:21:50.108 30923-31581/com.redacted.redacted E/CRASH:     x16 000000768b6fcbc0  x17 000000768b6d7680  x18 00000073ba370000  x19 00000000000078cb
2022-09-29 18:21:50.108 30923-31581/com.redacted.redacted E/CRASH:     x20 0000000000007b5d  x21 00000000ffffffff  x22 ffffff80ffffffc8  x23 0000007554be16c0
2022-09-29 18:21:50.108 30923-31581/com.redacted.redacted E/CRASH:     x24 0000007554be15a0  x25 0000007554be15e0  x26 0000000000000019  x27 00000000000fc000
2022-09-29 18:21:50.108 30923-31581/com.redacted.redacted E/CRASH:     x28 0000007554ae9000  x29 0000007554be14f0
2022-09-29 18:21:50.108 30923-31581/com.redacted.redacted E/CRASH:     sp  0000007554be1450  lr  000000768b68799c  pc  000000768b6879c8
2022-09-29 18:21:50.108 30923-31581/com.redacted.redacted E/CRASH: backtrace:
2022-09-29 18:21:50.108 30923-31581/com.redacted.redacted E/CRASH:       #00 pc 00000000000799c8  /apex/com.android.runtime/lib64/bionic/libc.so (abort+168) (BuildId: bbbdeb7c87c74f1491f92c6e605095b0)
2022-09-29 18:21:58.682 30923-30923/com.redacted.redacted E/CRASH: other thread is trapped; signum = 6
2022-09-29 18:21:58.684 30923-30923/com.redacted.redacted E/InputEventReceiver: Exception dispatching input event.
2022-09-29 18:21:58.684 30923-30923/com.redacted.redacted E/MessageQueue-JNI: Exception in MessageQueue callback: handleReceiveCallback
2022-09-29 18:21:58.685 30923-30923/com.redacted.redacted E/MessageQueue-JNI: java.lang.Error: *** *** *** *** *** *** *** *** *** *** *** *** *** *** *** ***
    Version '2021.3.5f1 (40eb3a945986)', Build type 'Development', Scripting Backend 'il2cpp', CPU 'arm64-v8a'
    Build fingerprint: 'OnePlus/OnePlus9/OnePlus9:12/RKQ1.211119.001/R.202208121853:user/release-keys'
    Revision: '0'
    ABI: 'arm64'
    Timestamp: 2022-09-29 18:21:50+0300
    pid: 30923, tid: 31581, name: Thread-24  >>> com.redacted.redacted <<<
    uid: 10399
    signal 6 (SIGABRT), code -1 (SI_QUEUE), fault addr --------
        x0  0000000000000000  x1  0000000000007b5d  x2  0000000000000006  x3  0000007554be1470
        x4  736f646277641f73  x5  736f646277641f73  x6  736f646277641f73  x7  7f7f7f7f7f7f7f7f
        x8  00000000000000f0  x9  b9efe8a004ca2f92  x10 0000000000000000  x11 ffffff80fffffbdf
        x12 0000000000000001  x13 00003215055312db  x14 00373df0a02a9fa1  x15 0000000034155555
        x16 000000768b6fcbc0  x17 000000768b6d7680  x18 00000073ba370000  x19 00000000000078cb
        x20 0000000000007b5d  x21 00000000ffffffff  x22 ffffff80ffffffc8  x23 0000007554be16c0
        x24 0000007554be15a0  x25 0000007554be15e0  x26 0000000000000019  x27 00000000000fc000
        x28 0000007554ae9000  x29 0000007554be14f0
        sp  0000007554be1450  lr  000000768b68799c  pc  000000768b6879c8
    
    backtrace:
          #00 pc 00000000000799c8  /apex/com.android.runtime/lib64/bionic/libc.so (abort+168) (BuildId: bbbdeb7c87c74f1491f92c6e605095b0)
    
        at libc.abort(abort:168)
2022-09-29 18:21:58.686 30923-30923/com.redacted.redacted E/AndroidRuntime: FATAL EXCEPTION: main
    Process: com.redacted.redacted, PID: 30923
    java.lang.Error: *** *** *** *** *** *** *** *** *** *** *** *** *** *** *** ***
    Version '2021.3.5f1 (40eb3a945986)', Build type 'Development', Scripting Backend 'il2cpp', CPU 'arm64-v8a'
    Build fingerprint: 'OnePlus/OnePlus9/OnePlus9:12/RKQ1.211119.001/R.202208121853:user/release-keys'
    Revision: '0'
    ABI: 'arm64'
    Timestamp: 2022-09-29 18:21:50+0300
    pid: 30923, tid: 31581, name: Thread-24  >>> com.redacted.redacted <<<
    uid: 10399
    signal 6 (SIGABRT), code -1 (SI_QUEUE), fault addr --------
        x0  0000000000000000  x1  0000000000007b5d  x2  0000000000000006  x3  0000007554be1470
        x4  736f646277641f73  x5  736f646277641f73  x6  736f646277641f73  x7  7f7f7f7f7f7f7f7f
        x8  00000000000000f0  x9  b9efe8a004ca2f92  x10 0000000000000000  x11 ffffff80fffffbdf
        x12 0000000000000001  x13 00003215055312db  x14 00373df0a02a9fa1  x15 0000000034155555
        x16 000000768b6fcbc0  x17 000000768b6d7680  x18 00000073ba370000  x19 00000000000078cb
        x20 0000000000007b5d  x21 00000000ffffffff  x22 ffffff80ffffffc8  x23 0000007554be16c0
        x24 0000007554be15a0  x25 0000007554be15e0  x26 0000000000000019  x27 00000000000fc000
        x28 0000007554ae9000  x29 0000007554be14f0
        sp  0000007554be1450  lr  000000768b68799c  pc  000000768b6879c8
    
    backtrace:
          #00 pc 00000000000799c8  /apex/com.android.runtime/lib64/bionic/libc.so (abort+168) (BuildId: bbbdeb7c87c74f1491f92c6e605095b0)
    
        at libc.abort(abort:168)
2022-09-29 18:21:58.705 30923-30923/com.redacted.redacted I/Process: Sending signal. PID: 30923 SIG: 9
```

**Comment by Gallardo994** [comment]
Hello. I am the guy who reported this issue on the forums, so if you need testing if any potential fix works - I am glad to do so. Thanks.

**Comment by olokobayusuf** [comment]
Just wanted to update you @Gallardo994 , we're actively working on this. Sit tight!

**Comment by olokobayusuf** [comment]
Same crash from different dev:
```
backtrace:
  #00  pc 0x000000000004e574  /apex/com.android.runtime/lib64/bionic/libc.so (abort+180)
  #01  pc 0x000000000008a74c  /data/app/~~kqaE2z1n94pGbaaF7GL0-w==/xxx.xxx.xxx-dqG2tmWqoQbE81w3br5flQ==/lib/arm64/libNatMLHub.so (std::set_unexpected(void (*)()))
  #02  pc 0x000000000008a8a4  /data/app/~~kqaE2z1n94pGbaaF7GL0-w==/xxx.xxx.xxx-dqG2tmWqoQbE81w3br5flQ==/lib/arm64/libNatMLHub.so
  #03  pc 0x0000000000086794  /data/app/~~kqaE2z1n94pGbaaF7GL0-w==/xxx.xxx.xxx-dqG2tmWqoQbE81w3br5flQ==/lib/arm64/libNatMLHub.so
  #04  pc 0x0000000000085dc0  /data/app/~~kqaE2z1n94pGbaaF7GL0-w==/xxx.xxx.xxx-dqG2tmWqoQbE81w3br5flQ==/lib/arm64/libNatMLHub.so (__cxa_get_exception_ptr)
  #05  pc 0x0000000000085d1c  /data/app/~~kqaE2z1n94pGbaaF7GL0-w==/xxx.xxx.xxx-dqG2tmWqoQbE81w3br5flQ==/lib/arm64/libNatMLHub.so
  #06  pc 0x000000000007a6dc  /data/app/~~kqaE2z1n94pGbaaF7GL0-w==/xxx.xxx.xxx-dqG2tmWqoQbE81w3br5flQ==/lib/arm64/libNatMLHub.so (operator new(unsigned long, std::nothrow_t const&))
  #07  pc 0x0000000000020efc  /data/app/~~kqaE2z1n94pGbaaF7GL0-w==/xxx.xxx.xxx-dqG2tmWqoQbE81w3br5flQ==/lib/arm64/libNatDevice.so
  #08  pc 0x00000000000b0838  /apex/com.android.runtime/lib64/bionic/libc.so (__pthread_start(void*)+64)
  #09  pc 0x00000000000505d0  /apex/com.android.runtime/lib64/bionic/libc.so (__start_thread+64)
```

**Comment by marvpaul** [comment]
Experience the same, also got a crash on startup when trying to access NatDevice's microphone.

**Comment by olokobayusuf** [comment]
[Fixed in 1.3.0](https://github.com/natmlx/NatDevice/releases/tag/1.3.0).
